### PR TITLE
RE-1163 Replace Travis CI with standard jobs

### PR DESCRIPTION
**Pull request by git-harry**
PR jobs have been created in rpc-gating to run the tests that were the
responsibility of Travis CI, they are:

- ansible-lint
- ansible-syntax
- bashate
- pep8
- releasenotes

The `run` and `post` scripts now check the standard environment variable
`RE_JOB_ACTION`, to determine the type of build that is running. If the
variable is set to `"tox-test"` a tox environment is requested and the
specific one is defined in `RE_JOB_SCENARIO` otherwise the scripts
default to their original behaviour of deploying RPCO.

Issue: [RE-1163](https://rpc-openstack.atlassian.net/browse/RE-1163)

### CODE REVIEW

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_xenial_swift_deploy

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_pep8_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_container_ansible-lint_tox-test
recheck_cit_container_ansible-syntax_tox-test
recheck_cit_container_bashate_tox-test
recheck_cit_container_pep8_tox-test
recheck_cit_container_releasenotes_tox-test

**Comment by git-harry** [comment]
recheck_cit_release

**Comment by major** [comment]
@git-harry Should we be ignoring the results of `continuous-integration/travis-ci` on this PR?

**Comment by git-harry** [comment]
@major this pull request shouldn't be merged until https://github.com/rcbops/rpc-gating/pull/554 is merged. Once that is done I will be able to update the list of required jobs and remove the travis-ci entry. This pull request is removing support for Travis CI and so the check can be ignored.

File: gating/pre_merge_test/post [hunk 1]
```diff
@@ -5,53 +5,8 @@
...
-    CONTAINER_PID=$(lxc-info -p -n ${CONTAINER_NAME} | awk '{print $2}')
-    CONTAINER_ROOT_DIR="/proc/${CONTAINER_PID}/root/"
-    CONTAINER_ETC_DIR="${CONTAINER_ROOT_DIR}/etc/"
-    if echo ${CONTAINER_NAME} | grep "utility"; then
-      find "${CONTAINER_ROOT_DIR}" -type f -name 'tempest_results.xml' -exec cp {} ${RE_HOOK_RESULT_DIR}/ \;
-    fi
-    ${RSYNC_ETC_CMD} ${CONTAINER_ETC_DIR} "${RE_HOOK_ARTIFACT_DIR}/etc/openstack/${CONTAINER_NAME}/" || true
-  done
+if [ $RE_JOB_ACTION != "tox-test" ]; then
+  bash -c "$(readlink -f $(dirname ${0})/post_deploy.sh)"
```
Comment (hughsaunders) [inline gating/pre_merge_test/post]: Why is readlink needed here? Why not `bash $(dirname ${0})/post_deploy.sh` ?

Comment (git-harry) [reply gating/pre_merge_test/post]: @hughsaunders I was copying the existing pattern in the repo.

**Pull request merged**